instance/models/instance.py

```diff
@@ -207,52 +205,61 @@ def _sort_log_entries(server_logs, instance_logs):
         next_server_log_entry = partial(next, server_logs, None)
         next_instance_log_entry = partial(next, instance_logs, None)
 
-        log = []
         instance_log_entry = next_instance_log_entry()
         server_log_entry = next_server_log_entry()
 
         while instance_log_entry is not None and server_log_entry is not None:
             if server_log_entry.created < instance_log_entry.created:
-                log.append(server_log_entry)
+                yield server_log_entry
                 server_log_entry = next_server_log_entry()
             else:
-                log.append(instance_log_entry)
+                yield instance_log_entry
                 instance_log_entry = next_instance_log_entry()
 
         while instance_log_entry is not None:
-            log.append(instance_log_entry)
+            yield instance_log_entry
             instance_log_entry = next_instance_log_entry()
 
         while server_log_entry is not None:
-            log.append(server_log_entry)
+            yield server_log_entry
             server_log_entry = next_server_log_entry()
 
-        return log
-
-    def _get_log_entries(self, level_list=None):
+    def _get_log_entries(self, level_list=None, limit=None):
         """
         Return the list of log entry instances for the instance and its current active server,
-        optionally filtering by logging level.
+        optionally filtering by logging level. If a limit is given, only the latest records are
+        returned.
         """
         # TODO: Filter out log entries for which the user doesn't have view rights
         server_log_entry_set = self._current_server.log_entry_set
         if level_list:
             server_log_entry_set = server_log_entry_set.filter(level__in=level_list)
-        server_log_entry_set = server_log_entry_set.order_by('pk').iterator()
+        server_log_entry_set = server_log_entry_set.order_by('pk')
+        if limit:
+            server_log_entry_set = reversed(server_log_entry_set.reverse()[:limit])
+        else:
+            server_log_entry_set = server_log_entry_set.iterator()
 
         instance_log_entry_set = self.log_entry_set
         if level_list:
             instance_log_entry_set = instance_log_entry_set.filter(level__in=level_list)
-        instance_log_entry_set = instance_log_entry_set.order_by('pk').iterator()
+        instance_log_entry_set = instance_log_entry_set.order_by('pk')
+        if limit:
+            instance_log_entry_set = reversed(instance_log_entry_set.reverse()[:limit])
+        else:
+            instance_log_entry_set = instance_log_entry_set.iterator()
 
-        return Instance._sort_log_entries(server_log_entry_set, instance_log_entry_set)
+        entries = list(self._sort_log_entries(server_log_entry_set, instance_log_entry_set))
```

Comment:
The problem is that we want the *last* X entries. We could do that with `itertools.islice` by iterating backwards and then reversing the order.